packages/grafana-ui/src/components/Monaco/CodeEditor.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { withTheme } from '../../themes';
+import { Themeable } from '../../types';
+import { MonacoEditorProps } from 'react-monaco-editor';
+import { KeyCode, editor, KeyMod } from 'monaco-editor/esm/vs/editor/editor.api';
+import { useAsyncDependency } from '../../utils/useAsyncDependency';
+import { ErrorWithStack, LoadingPlaceholder } from '..';
+
+export type CodeEditorChangeHandler = (value: string) => void;
+
+const MonacoEditor: React.FC<MonacoEditorProps> = props => {
+  const { loading, error, dependency } = useAsyncDependency(
+    import(/* webpackChunkName: "react-monaco-editor" */ 'react-monaco-editor')
```

Comment:
does this not work?

Reply:
@torkelo it does. But monaco editor core does not become a part of this asynch chunk, even though it's imported by react-monaco editor. Hence the split chunk for the editor.